hastexo_guacamole_client/consumers.py

```diff
@@ -33,20 +33,24 @@ async def connect(self):
 
         self.read_only = bool(strtobool(params.get('read_only')[0]))
 
+        connection_args = {
+            'protocol': stack.protocol,
+            'width': params.get('width', [1024])[0],
+            'height': params.get('height', [768])[0],
+            'hostname': stack.ip,
+            'port': params.get('port', [default_port])[0],
+            'username': stack.user,
+            'password': stack.password,
+            'private-key': stack.key,
+            'color-scheme': settings.get("terminal_color_scheme"),
+            'font-name': settings.get("terminal_font_name"),
+            'font-size': settings.get("terminal_font_size"),
+        }
+        if stack.protocol == 'rdp':
+            connection_args['resize-method'] = 'reconnect'
+
         self.client = GuacamoleClient(guacd_hostname, guacd_port)
-        self.client.handshake(
-            protocol=stack.protocol,
-            width=params.get('width', [1024])[0],
-            height=params.get('height', [768])[0],
-            hostname=stack.ip,
-            port=params.get('port', [default_port])[0],
-            username=stack.user,
-            password=stack.password,
-            private_key=stack.key,
-            color_scheme=settings.get("terminal_color_scheme"),
-            font_name=settings.get("terminal_font_name"),
-            font_size=settings.get("terminal_font_size"),
-        )
+        self.client.handshake(connection_args)
```

Comment:
I may be making a fool of myself here because something might have changed in Python ages ago, and I may have missed it, but looking at [`GuacamoleClient.handshake()`](https://github.com/mohabusama/pyguacamole/blob/master/guacamole/client.py#L138) shouldn't this line look like so?

```suggestion
        self.client.handshake(**connection_args)
```

Reply:
Yes, that was a typo. But, I believe we don't need to split the arguments between different protocols at all, SSH does not have the "resize-method" argument and will simply ignore it.